sv/message.go

```diff
@@ -202,7 +203,13 @@ func (p MessageProcessorImpl) Format(msg CommitMessage) (string, string, string)
 
 // Parse a commit message.
 func (p MessageProcessorImpl) Parse(subject, body string) CommitMessage {
-	commitType, scope, description, hasBreakingChange := parseSubjectMessage(subject)
+	preparedSubject, prepError := p.prepareHeader(subject)
+
+	if prepError != nil {
+	    fmt.Println(prepError)
```

Comment:
I think you forgot to return the error here so it can be handled outside `sv` package..

You will need to change Parse function to return an error like this (don't forget to change the `MessageProcessor` interface as well):

```go
// Parse a commit message.
func (p MessageProcessorImpl) Parse(subject, body string) (CommitMessage, error) {
	preparedSubject, err := p.prepareHeader(subject)
	if err != nil {
		return CommitMessage{}, err
	}
(...)
```
and then , there is a minor changes that will be needed on `git.go` to be able to return the error on `func (g GitImpl) Log(lr LogRange) ([]GitCommitLog, error)`

Reply:
Hi, I added and changed the error behavior. The behaviour is now to ignore commits which are not parsable. Also if the header selector does not get a match it will default to the message instead, so the header selector is not a must match now. 

Especially for the validate functionality this is helpful I guess. What do you think?

Reply:
Nevermind. I changed this to be a must-match. Such a behaviour is then implicitly configurable through the regex pattern.

Reply:
do you mean by adding the prefix as optional?

eg: `(Merged PR (\\d+): )?(?P<header>.*)`

Reply:
Yes, exactly